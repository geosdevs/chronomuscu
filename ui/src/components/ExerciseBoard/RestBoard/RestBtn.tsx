import { secondsToPrettyString } from "../../../helpers/time";
import React, { useContext } from "react";
import { ExerciseBoardRestBtnClickContext } from "../ExerciseBoard";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faHourglass } from "@fortawesome/free-solid-svg-icons";

type RestBtnProps = {
  restSeconds: number
  children?: React.ReactNode
  readOnly: boolean
};

export default function RestBtn({ restSeconds, readOnly }: RestBtnProps) {
  const onRestBtnClick = useContext(ExerciseBoardRestBtnClickContext);
  // let bgColor;
  
  // if (restSeconds >= 60 && restSeconds < 120) {
  //   bgColor = "bg-indigo-700";
  // } else if (restSeconds >= 120) {
  //   bgColor = "bg-indigo-800";
  // } else {
  //   bgColor = 'bg-indigo-600';
  // }

  return (
      <button
        disabled={readOnly}
        className={`m-1 rounded-full text-white 
        hover:scale-110 hover:bg-ecru hover:text-gunmetal
        bg-bluemunsell py-1 px-2 md:p-3 md:px-3 transition  
        leading-5 focus:outline-none focus:ring disabled:bg-gray-400`}
        onClick={() => {
          if (typeof onRestBtnClick === "function") {
            onRestBtnClick(restSeconds);
          }
        }}
      >
        <FontAwesomeIcon icon={faHourglass} size="xs"></FontAwesomeIcon>
        <span className="mx-1">{secondsToPrettyString(restSeconds)}</span>
      </button>
  );
}
