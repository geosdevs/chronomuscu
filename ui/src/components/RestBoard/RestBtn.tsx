import { secondsToPrettyString } from "../../functions/time";

type RestBtnProps = {
    restSeconds: number,
    children?: React.ReactNode
}

export default function RestBtn({restSeconds}: RestBtnProps) {
    return (
        <div>
            <button>{secondsToPrettyString(restSeconds)}</button>
        </div>
    )
}