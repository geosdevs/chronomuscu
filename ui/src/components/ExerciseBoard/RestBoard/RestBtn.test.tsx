import renderer from 'react-test-renderer';
import RestBtn from './RestBtn';

describe('RestBtn', () => {
  it('should render', () => {
    const restTimer = 35;
    const component = renderer.create((
      <RestBtn
      key={`rest-seconds-${restTimer}`}
      restSeconds={restTimer}
      readOnly={false}
    ></RestBtn>
    ));
    let tree = component.toTree();

    expect(tree).toMatchSnapshot();
  });

  it('should render readonly', () => {
    const restTimer = 35;
    const component = renderer.create((
      <RestBtn
      key={`rest-seconds-${restTimer}`}
      restSeconds={restTimer}
      readOnly={true}
    ></RestBtn>
    ));
    let tree = component.toTree();

    expect(tree).toMatchSnapshot();
  })
});