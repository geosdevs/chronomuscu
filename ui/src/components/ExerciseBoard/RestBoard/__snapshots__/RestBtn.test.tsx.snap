// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestBtn should render 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "readOnly": false,
    "restSeconds": 35,
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FontAwesomeIcon
          beat={false}
          beatFade={false}
          border={false}
          bounce={false}
          className=""
          fade={false}
          fixedWidth={false}
          flip={false}
          icon={
            Object {
              "icon": Array [
                384,
                512,
                Array [
                  9203,
                  62032,
                  "hourglass-empty",
                ],
                "f254",
                "M0 32C0 14.3 14.3 0 32 0H64 320h32c17.7 0 32 14.3 32 32s-14.3 32-32 32V75c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1v11c17.7 0 32 14.3 32 32s-14.3 32-32 32H320 64 32c-17.7 0-32-14.3-32-32s14.3-32 32-32V437c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75V64C14.3 64 0 49.7 0 32zM96 64V75c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9V64H96zm0 384H288V437c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9v11z",
              ],
              "iconName": "hourglass",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          maskId={null}
          pull={null}
          pulse={false}
          rotation={null}
          shake={false}
          size="xs"
          spin={false}
          spinPulse={false}
          spinReverse={false}
          swapOpacity={false}
          symbol={false}
          title=""
          titleId={null}
          transform={null}
        />,
        <span
          className="mx-1"
        >
          35s
        </span>,
      ],
      "className": "m-1 rounded-full text-white  enabled:hover:scale-110 hover:bg-ecru hover:text-white bg-chinarose py-1 px-2 md:p-3 md:px-3 transition   leading-5 focus:outline-none disabled:bg-gray-400",
      "disabled": false,
      "onClick": [Function],
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            d="M0 32C0 14.3 14.3 0 32 0H64 320h32c17.7 0 32 14.3 32 32s-14.3 32-32 32V75c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1v11c17.7 0 32 14.3 32 32s-14.3 32-32 32H320 64 32c-17.7 0-32-14.3-32-32s14.3-32 32-32V437c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75V64C14.3 64 0 49.7 0 32zM96 64V75c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9V64H96zm0 384H288V437c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9v11z"
            fill="currentColor"
            style={Object {}}
          />,
          "className": "svg-inline--fa fa-hourglass fa-xs ",
          "data-icon": "hourglass",
          "data-prefix": "fas",
          "focusable": "false",
          "role": "img",
          "style": Object {},
          "viewBox": "0 0 384 512",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "d": "M0 32C0 14.3 14.3 0 32 0H64 320h32c17.7 0 32 14.3 32 32s-14.3 32-32 32V75c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1v11c17.7 0 32 14.3 32 32s-14.3 32-32 32H320 64 32c-17.7 0-32-14.3-32-32s14.3-32 32-32V437c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75V64C14.3 64 0 49.7 0 32zM96 64V75c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9V64H96zm0 384H288V437c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9v11z",
              "fill": "currentColor",
              "style": Object {},
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": "35s",
          "className": "mx-1",
        },
        "rendered": Array [
          "35s",
        ],
        "type": "span",
      },
    ],
    "type": "button",
  },
  "type": [Function],
}
`;

exports[`RestBtn should render readonly 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "readOnly": true,
    "restSeconds": 35,
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FontAwesomeIcon
          beat={false}
          beatFade={false}
          border={false}
          bounce={false}
          className=""
          fade={false}
          fixedWidth={false}
          flip={false}
          icon={
            Object {
              "icon": Array [
                384,
                512,
                Array [
                  9203,
                  62032,
                  "hourglass-empty",
                ],
                "f254",
                "M0 32C0 14.3 14.3 0 32 0H64 320h32c17.7 0 32 14.3 32 32s-14.3 32-32 32V75c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1v11c17.7 0 32 14.3 32 32s-14.3 32-32 32H320 64 32c-17.7 0-32-14.3-32-32s14.3-32 32-32V437c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75V64C14.3 64 0 49.7 0 32zM96 64V75c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9V64H96zm0 384H288V437c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9v11z",
              ],
              "iconName": "hourglass",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          maskId={null}
          pull={null}
          pulse={false}
          rotation={null}
          shake={false}
          size="xs"
          spin={false}
          spinPulse={false}
          spinReverse={false}
          swapOpacity={false}
          symbol={false}
          title=""
          titleId={null}
          transform={null}
        />,
        <span
          className="mx-1"
        >
          35s
        </span>,
      ],
      "className": "m-1 rounded-full text-white  enabled:hover:scale-110 hover:bg-ecru hover:text-white bg-chinarose py-1 px-2 md:p-3 md:px-3 transition   leading-5 focus:outline-none disabled:bg-gray-400",
      "disabled": true,
      "onClick": [Function],
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            d="M0 32C0 14.3 14.3 0 32 0H64 320h32c17.7 0 32 14.3 32 32s-14.3 32-32 32V75c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1v11c17.7 0 32 14.3 32 32s-14.3 32-32 32H320 64 32c-17.7 0-32-14.3-32-32s14.3-32 32-32V437c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75V64C14.3 64 0 49.7 0 32zM96 64V75c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9V64H96zm0 384H288V437c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9v11z"
            fill="currentColor"
            style={Object {}}
          />,
          "className": "svg-inline--fa fa-hourglass fa-xs ",
          "data-icon": "hourglass",
          "data-prefix": "fas",
          "focusable": "false",
          "role": "img",
          "style": Object {},
          "viewBox": "0 0 384 512",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "d": "M0 32C0 14.3 14.3 0 32 0H64 320h32c17.7 0 32 14.3 32 32s-14.3 32-32 32V75c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1v11c17.7 0 32 14.3 32 32s-14.3 32-32 32H320 64 32c-17.7 0-32-14.3-32-32s14.3-32 32-32V437c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75V64C14.3 64 0 49.7 0 32zM96 64V75c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9V64H96zm0 384H288V437c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9v11z",
              "fill": "currentColor",
              "style": Object {},
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": "35s",
          "className": "mx-1",
        },
        "rendered": Array [
          "35s",
        ],
        "type": "span",
      },
    ],
    "type": "button",
  },
  "type": [Function],
}
`;
